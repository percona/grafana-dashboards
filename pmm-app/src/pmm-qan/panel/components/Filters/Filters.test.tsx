// Just a stub test
// import { Filters } from './Filters';
import React from 'react';

jest.mock('antd/es/input/Search', () => () => <></>);

jest.mock('shared/components/helpers/notification-manager', () => () => ({}));

// const MOCK_FILTERS = {
//   az: {
//     name: [
//       {
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   client_host: {
//     name: [
//       {
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   cluster: {
//     name: [
//       {
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   container_id: {
//     name: [
//       {
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   container_name: {
//     name: [
//       {
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   database: {
//     name: [
//       {
//         value: 'postgres',
//         main_metric_percent: 0.99822205,
//         main_metric_per_sec: 0.0028915745,
//         checked: false,
//       },
//       {
//         value: 'pmm-managed',
//         main_metric_percent: 0.001778005,
//         main_metric_per_sec: 0.000005150391,
//         checked: true,
//       },
//     ],
//   },
//   environment: {
//     name: [
//       {
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   machine_id: {
//     name: [
//       {
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   node_id: {
//     name: [
//       {
//         value: 'pmm-server',
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   node_model: {
//     name: [
//       {
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   node_name: {
//     name: [
//       {
//         value: 'pmm-server',
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   node_type: {
//     name: [
//       {
//         value: 'generic',
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   region: {
//     name: [
//       {
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   replication_set: {
//     name: [
//       {
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   schema: {
//     name: [
//       {
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   service_id: {
//     name: [
//       {
//         value: '/service_id/1b879be7-cf43-487a-82f0-09143f8d914b',
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   service_name: {
//     name: [
//       {
//         value: 'pmm-server-postgresql',
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   service_type: {
//     name: [
//       {
//         value: 'postgresql',
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
//   username: {
//     name: [
//       {
//         value: 'pmm-managed',
//         main_metric_percent: 1,
//         main_metric_per_sec: 0.000005150391,
//       },
//     ],
//   },
// };

xdescribe('Filters test', () => {
  xit('Renders correct with right props', () => {
    // const component = renderer.create(
    //   <Filters
    //     groups={FILTERS_GROUPS}
    //     filters={MOCK_FILTERS}
    //     form={{ reset: () => {} }}
    //     labels={{}}
    //     dispatch={() => {}}
    //   />
    // );
    //
    // const tree = component.toJSON();
    //
    // expect(tree).toMatchSnapshot();
  });
});
