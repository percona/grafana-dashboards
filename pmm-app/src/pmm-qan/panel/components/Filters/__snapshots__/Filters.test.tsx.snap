// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters test Renders correct with right props 1`] = `
<div>
  <div
    className="css-1lkrf7x"
  >
    <h5
      className="css-1n9a2zn"
    >
      Filters
    </h5>
    <button
      className="ant-btn css-x930nf ant-btn-link"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        Show Selected
      </span>
    </button>
    <button
      className="ant-btn css-ym22kl ant-btn-link"
      disabled={false}
      id="reset-all-filters"
      onClick={[Function]}
      type="button"
    >
      <span>
        Reset All
      </span>
    </button>
  </div>
  <div />
</div>
`;
